app/scripts/modules/core/src/pipeline/executions/execution/ExecutionMarkerInformationPopover.tsx

```diff
@@ -0,0 +1,360 @@
+          <div className="information-details">
+            <div className="pipeline-name bottom-margin">
+              {pipelineConfigDetails ? pipelineConfigDetails.name : '-'}
+            </div>
+            <div style={{ fontWeight: 'bold' }}>PIPELINE</div>
+            <div className="bottom-margin">
+              <div>{`[${executionDetails.authentication.user}] (${executionDetails.user})`}</div>
+              <div>{relativeTime(executionDetails.startTime || executionDetails.buildTime)}</div>
+              <div>
+                Status: <span className="status">{executionDetails.status}</span> by parent pipeline{' '}
+                {timestamp(executionDetails.trigger.parentExecution.buildTime)}
+              </div>
+            </div>
+            <div className="bottom-margin parameters">
+              <span
+                className="clickable"
+                onClick={event => {
+                  this.toggleParameters(event.target);
+                }}
+              >
+                <i className="fa fa-chevron-right"></i> View all Parameters (
+                {Object.keys(executionDetails.trigger.parameters).length})
+              </span>
+              <div className="view-parameters closed">
+                {Object.keys(executionDetails.trigger.parameters).map((key: string, index: number) => (
+                  <span key={`${key}-${index}`}>
+                    {key}: {executionDetails.trigger.parameters[key]}
+                  </span>
+                ))}
+              </div>
+            </div>
+            {stageDetails && (
+              <div className="bottom-margin information-stage-details">
+                <div>STAGE DETAILS</div>
+                <div>Name: {stageDetails.name}</div>
+                <div>Duration: {duration(stageDetails.endTime - stageDetails.startTime)}</div>
+                <div>Exception: {stageDetails.getErrorMessage || 'No message available'}</div>
+                <div>
+                  Pipeline Execution History <span className="information-history-note">(Decending order)</span>
+                  <table className="information-pipeline-execution-history">
+                    <thead>
+                      <tr>
+                        <th></th>
+                        <th>APPLICATION</th>
+                        <th>PIPELINE NAME</th>
+                        <th>STAGE</th>
+                        <th>STATUS</th>
+                        <th>DURATION</th>
+                      </tr>
+                    </thead>
+                    <tbody className="information-section">
+                      {this.allExecutions.map((item: any, index: number) => {
+                        return (
+                          <tr
+                            key={`execution-history-for-${item.execution.id}-${item.stageId}`}
+                            onClick={() => {
+                              (document.querySelector(`#stage-${item.stageId}`) as HTMLAnchorElement).click();
```

Comment:
Whoah, what's going on here? Any other way to handle?

Reply:
you wanted the the angular router to handle links but i need an entire row to be clicked on i tried to wrap an entire row in that, i know tables sucks but they a really are still the go to with sizing. i can play around with something else. since we are using the angular ui router link when a row is clicked i just search for that and click it.